pkger/clone_resource.go

```diff
@@ -31,6 +37,300 @@ func (r ResourceToClone) OK() error {
 	return nil
 }
 
+var kindPriorities = map[Kind]int{
+	KindLabel:                         1,
+	KindBucket:                        2,
+	KindCheck:                         3,
+	KindCheckDeadman:                  4,
+	KindCheckThreshold:                5,
+	KindNotificationEndpoint:          6,
+	KindNotificationEndpointHTTP:      7,
+	KindNotificationEndpointPagerDuty: 8,
+	KindNotificationEndpointSlack:     9,
+	KindNotificationRule:              10,
+	KindTask:                          11,
+	KindVariable:                      12,
+	KindDashboard:                     13,
+	KindTelegraf:                      14,
+}
+
+type exportKey struct {
+	orgID influxdb.ID
+	name  string
+	kind  Kind
+}
+
+func newExportKey(orgID influxdb.ID, k Kind, name string) exportKey {
+	return exportKey{
+		orgID: orgID,
+		name:  name,
+		kind:  k,
+	}
+}
+
+type exportCoordinator struct {
+	bucketSVC   influxdb.BucketService
+	checkSVC    influxdb.CheckService
+	dashSVC     influxdb.DashboardService
+	labelSVC    influxdb.LabelService
+	endpointSVC influxdb.NotificationEndpointService
+	ruleSVC     influxdb.NotificationRuleStore
+	taskSVC     influxdb.TaskService
+	teleSVC     influxdb.TelegrafConfigStore
+	varSVC      influxdb.VariableService
+
+	mObjects map[exportKey]Object
+}
+
+func (ex *exportCoordinator) Export(ctx context.Context, resourcesToClone []ResourceToClone, labelNames ...string) error {
+	cloneAssFn, err := ex.resourceCloneAssociationsGen(ctx, labelNames...)
+	if err != nil {
+		return err
+	}
+
+	resourcesToClone = uniqResourcesToClone(resourcesToClone)
+	sort.Slice(resourcesToClone, func(i, j int) bool {
+		iName, jName := resourcesToClone[i].Name, resourcesToClone[j].Name
+		iKind, jKind := resourcesToClone[i].Kind, resourcesToClone[j].Kind
+
+		if iKind.is(jKind) {
+			return iName < jName
+		}
+		return kindPriorities[iKind] < kindPriorities[jKind]
+	})
+
+	for _, r := range resourcesToClone {
+		err := ex.resourceCloneToKind(ctx, r, cloneAssFn)
+		if err != nil {
+			return internalErr(fmt.Errorf("failed to clone resource: resource_id=%s resource_kind=%s err=%q", r.ID, r.Kind, err))
```

Comment:
Internal Err is the default code, you can just use fmt.Errorf

Reply:
I want to be explicit about using an `influxdb.Error` here. There is associated behavior with that type. That behavior I want the service to guarantee it handles so that consumers of this service have nothing but a error code check to signal how they will handle it.